code/crates/core-types/src/value.rs

```diff
@@ -130,8 +132,20 @@ impl ValuePayload {
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub enum ValueOrigin {
     /// Synchronization protocol
-    Sync,
+    Sync(PeerId),
```

Comment:
We should maybe add at some point the peer to `height_per_request_id` and `pending_value_requests` in sync so we wouldn't need to keep it here.

Reply:
Yes definitely a better of doing this

Reply:
Will do this in a follow-up PR.